kubernetes/windows/main.ps1

```diff
@@ -44,6 +44,61 @@ function Set-ProcessAndMachineEnvVariables($name, $value) {
     [System.Environment]::SetEnvironmentVariable($name, $value, "Machine")
 }
 
+function Set-AMAEnvironmentVariables {
+    Set-ProcessAndMachineEnvVariables "MONITORING_DATA_DIRECTORY" "C:\\opt\\windowsazuremonitoragent\\datadirectory"
+    Set-ProcessAndMachineEnvVariables "MONITORING_MCS_MODE" "1"
+    Set-ProcessAndMachineEnvVariables "MONITORING_ROLE_INSTANCE" "cloudAgentRoleInstanceIdentity"
+    Set-ProcessAndMachineEnvVariables "MA_RoleEnvironment_OsType" "Windows"
+    Set-ProcessAndMachineEnvVariables "MONITORING_VERSION" "2.0"
+    Set-ProcessAndMachineEnvVariables "MONITORING_ROLE" "cloudAgentRoleIdentity"
+    Set-ProcessAndMachineEnvVariables "MONITORING_IDENTITY" "use_ip_address"
+
+    $aksRegion = [System.Environment]::GetEnvironmentVariable("AKS_REGION", "process")
+    Set-ProcessAndMachineEnvVariables "MA_RoleEnvironment_Location" $aksRegion
+    Set-ProcessAndMachineEnvVariables "customRegion" $aksRegion
+
+    $aksResourceId = [System.Environment]::GetEnvironmentVariable("AKS_RESOURCE_ID", "process")
```

Comment:
not sure if we are doing this in Linux but is validating that these env values are not null needed here?